controllers/main.js

```diff
@@ -113,29 +117,38 @@ module.exports.sendPlayData = async (req, res, next) => {
 
 module.exports.updatePlayData = async (req, res, next) => {
   const sessionId = req.body.sessionId;
-  const pageNum = parseInt(req.body.pageNum);
-  const playSession = await PlaySession.findOne({ sessionId: sessionId });
-  const userGuess =
-    req.body.guess === playSession.sessionData[pageNum - 1].gameName;
-  const elemId = req.body.elemId;
-  await PlaySession.updateOne(
-    { sessionId: sessionId },
-    {
-      $set: {
-        "sessionData.$[elem].userGuess": userGuess,
-        "sessionData.$[elem].userGuessText": req.body.guess,
-      },
-    },
-    { arrayFilters: [{ "elem._id": elemId }] }
-  );
-  const playSessionData = await fetchPlaySessionData(req.user.username);
-  if (req.body.action === "back") {
-    res.redirect(`/play/${sessionId}/${pageNum - 1}`);
-  } else if (playSessionData.length === parseInt(req.body.pageNum)) {
+  const thisSession = await PlaySession.findOne({ sessionId: sessionId });
+  if (thisSession.sessionEnded) {
     res.redirect("/results");
+  } else {
+    const pageNum = parseInt(req.body.pageNum);
+    const playSession = await PlaySession.findOne({ sessionId: sessionId });
+    const userGuess =
+      req.body.guess === playSession.sessionData[pageNum - 1].gameName;
+    const elemId = req.body.elemId;
+    await PlaySession.updateOne(
+      { sessionId: sessionId },
+      {
+        $set: {
+          "sessionData.$[elem].userGuess": userGuess,
+          "sessionData.$[elem].userGuessText": req.body.guess,
+        },
+      },
+      { arrayFilters: [{ "elem._id": elemId }] }
+    );
+    const playSessionData = await fetchPlaySessionData(req.user.username);
+    if (req.body.action === "back") {
```

Comment:
Would be nice to have these action types declared separately somwhere.

```
const PLAY_ACTION_TYPES = {
  back: 'back'
}
```

Reply:
Could you elaborate a bit more on this one?